operator/pkg/helmreconciler/apply.go

```diff
@@ -230,3 +251,72 @@ func (h *HelmReconciler) serverSideApply(obj *unstructured.Unstructured) error {
 	}
 	return nil
 }
+
+func (h *HelmReconciler) analyzeWebhooks(whs object.K8sObjects) error {
+	// Get YAML string from K8sObject
+	var whYAMLs []string
+	whNameMap := make(map[string]bool)
+	for _, whObj := range whs {
+		yaml, err := whObj.YAML()
+		if err != nil {
+			return err
+		}
+		whYAMLs = append(whYAMLs, string(yaml))
+		whNameMap[whObj.Name] = true
+	}
+	if len(whYAMLs) == 0 {
+		return nil
+	}
+
+	globalI, ok := h.iop.Spec.Values["global"]
+	if !ok {
+		return fmt.Errorf("")
+	}
+	global, ok := globalI.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("")
+	}
+	istioNamespace, ok := global["istioNamespace"].(string)
+	if !ok {
+		return fmt.Errorf("")
+	}
+	sa := local.NewSourceAnalyzer(istioConfigSchema.MustGet(), analysis.Combine("webhook", &webhook.Analyzer{}),
+		resource.Namespace(h.iop.Spec.GetNamespace()), resource.Namespace(istioNamespace), nil, true, 30*time.Second)
+	var localWebhookYAMLReaders []local.ReaderSource
```

Comment:
We have a lot of code about extracting the webhook yamls. Would it be simpler to just pass all the YAMLs to the analyzer code? I believe it will just ignore resources it doesn't care about (non-webhooks)

Reply:
This actually make a lot of sense
I had the same concern that a lot of duplication of this logic is there in places (content.go and tag.go) 
Will work on this as well

Reply:
I looked at the repetitive implementations, It doesn't look like extracting webhooks is taking is the clutter, Just that procedure to get metadata like namespaces and converting into YAML strings is the issue. This is the result of having different ways to fetch data at different places in the code.
Ideally, We should wrap this up in a single function in utils somewhere and that might help us do away with some of this
But, It's not extracting webhooks.